pkg/select/blockdevice/select.go

```diff
@@ -63,6 +69,11 @@ func (c *Config) getCandidateDevices(bdList *apis.BlockDeviceList) (*apis.BlockD
 			if verifyDeviceType && bd.Spec.Details.DeviceType != c.ClaimSpec.DeviceType {
 				continue
 			}
+
+			if isBlockVolumeMatch(*c.ClaimSpec, bd.Spec, verifyVolumeMode, verifyFSType) {
```

Comment:
This is a good one.. to push the check into a seperate method. Can you also push the verify checks into that method. Change to have only two parameters - ClaimSpec and BDSpec. Since others can be derived within the function.  We can probably call the function as `verifyBlockDeviceMode`